### DISCUSSION TITLE
Displaying cursor location in both read&write mode

### DISCUSSION THREAD

**Discussion by yixin0829** [tamlok/vnote#1699] [Ideas]
I usually switch between write and read mode quite often using ^+t key binding but noticed sometimes the cursor location changes randomly if i switch to read mode, scroll up or down a bit so the cursor is out of the page, and then switch back to the write mode.

Is it possible to add a feature that allows us to see the cursor in read mode and navigate it using vim hjkl as well?

right now in read mode we can only use j and k key to move up and down but think we can replace them with ^+d and ^+u. and use hjkl for navigating the cursor instead?

**Comment by tamlok** [comment]
Which version of VNote?

**Reply by yixin0829** [reply]
version 3

**Reply by tamlok** [reply]
For v3.0, it already tries to map between the line number in edit mode and the paragraph element in read mode when switching the mode. It is hard to map them precisely after rendering Markdown to HTML.

When you switch from read mode into edit mode, VNote will try to get the line number of the first top paragraph element in current view port and jump to that line in edit mode.

**Comment by yixin0829** [comment]
yeah i noticed that but feel like it the mapping is still not precise enough as it usually maps the cursor to be near the top view window and it takes a bit of time to navigate the cursor to the exact location when switching back to the write mode when you do the switch frequently

Just putting the idea out there tho. it would be nice to have read/write mode switching as easy as switching b/w normal and insert mode in vim :)

**Reply by yixin0829** [reply]
also is it possible to have scrolloff option in VNote?

**Reply by tamlok** [reply]
There is a setting in the Settings to center cursor when on bottom, which is like the effect of scrolloff.

**Reply by yixin0829** [reply]
Got it. Thanks for pointing it out!
